proposals/gc/MVP.md

```diff
@@ -0,0 +1,272 @@
+# GC v1 Extensions
+
+
+## Language
+
+Based on reference types proposal.
+
+### Types
+
+#### Value Types
+
+* `eqref` is a new reference type
+  - `reftype ::= ... | eqref`
+
+* `ref <typeidx>` is a new reference type
+  - `reftype ::= ... | ref <typeidx>`
+  - `ref $t ok` iff `$t` is defined in the context
+
+* `int31ref` is a new reference type
+  - `reftype ::= ... | int31ref`
+
+
+#### Type Definitions
+
+* `deftype` is the new category of types that can occur as type definitions
+  - `deftype ::= <functype> | <structtype> | <arraytype>`
+  - `module ::= {..., types vec(<deftype>)}`
+
+* `structtype` describes a structure with statically indexed fields
+  - `structtype ::= struct <fieldtype>*`
+
+* `arraytype` describes an array with dynamically indexed fields
+  - `arraytype ::= array <fieldtype>`
```

Comment:
What about the older idea of giving `structtype`s a trailing, dynamic-length array tail?  I'd be surprised if multiple source-languages didn't need to store one or two extra bookkeeping fields to implement their source-language array, so it feels like a GC MVP thing and it seems like, with that ability, a separate `arraytype` isn't even needed.

Reply:
I agree that this would really extend language's abilities to find efficient representations for their data types. Engines can still treat structs without a tail specially.

Reply:
Fair enoug